core/src/main/java/com/linecorp/armeria/client/HttpClientBuilder.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client;
+
+import static java.util.Objects.requireNonNull;
+
+import java.net.URI;
+import java.util.function.Function;
+
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.SerializationFormat;
+import com.linecorp.armeria.common.SessionProtocol;
+
+/**
+ * Creates a new HTTP client that connects to the specified {@link URI} using the builder pattern.
+ * Use the factory methods in {@link HttpClient} if you do not have many options to override.
+ * Please refer to {@link ClientBuilder} for how decorators and HTTP headers are configured
+ */
+public final class HttpClientBuilder extends AbstractClientOptionsBuilder<HttpClientBuilder> {
+
+    private final URI uri;
+    private ClientFactory factory = ClientFactory.DEFAULT;
+
+    /**
+     * Creates a new instance.
+     *
+     * @throws IllegalArgumentException if the scheme of the uri is not one of the fields
+     *                                  in {@link SessionProtocol}
+     */
+    public HttpClientBuilder(String uri) {
+        this(URI.create(requireNonNull(uri, "uri")));
+    }
+
+    /**
+     * Creates a new instance.
+     *
+     * @throws IllegalArgumentException if the scheme of the uri is not one of the fields
+     *                                  in {@link SessionProtocol}
+     */
+    public HttpClientBuilder(URI uri) {
+        validateScheme(requireNonNull(uri, "uri").getScheme());
+        this.uri = URI.create(SerializationFormat.NONE + "+" + uri.toString());
+    }
+
+    private static void validateScheme(String scheme) {
+        for (SessionProtocol p : SessionProtocol.values()) {
+            if (scheme.equalsIgnoreCase(p.uriText())) {
+                return;
+            }
+        }
+        throw new IllegalArgumentException(
+                "the scheme should be one of h1, h1h, h2, h2c, http and https. input: " + scheme);
```

Comment:
`"scheme: " + scheme + " (expected: one of " + ImmutableList.copyOf(SessionProtocol.values()) + ')'`?